samcli/cli/types.py

```diff
@@ -232,6 +239,17 @@ def convert(self, value, param, ctx):
 
         return result
 
+    def _add_value(self, result: dict, key: str, new_value: str):
+        """
+        Add a given value to a given key in the result map.
+        """
+        if self.multiple_values_per_key:
+            if not result.get(key):
```

Comment:
nit: something like defaultdict(list) could be used here, not a blocker.

Reply:
The type of `result` would have to change based on whether `multiple_values_per_key` is set though, because if result is a `defaultdict(list)`, we can't do the normal `result[key] = value` if `multiple_values_per_key` is not set.

It's possible, but I think it's a bit more complexity than it's worth.